### PR TITLE
Fixes Torch GPU Test for Binomial and Beta Distribution

### PR DESCRIPTION
**Pull request by sampathweb**
Fixes #18920 for Torch on GPU.  Move to Device before doing math with another tensor that's using `convert_to_tensor`.

### CODE REVIEW

File: keras/backend/torch/random.py [hunk 1]
```diff
@@ -208,8 +208,10 @@ def gamma(shape, alpha, dtype=None, seed=None):
 def binomial(shape, counts, probabilities, dtype=None, seed=None):
     dtype = dtype or floatx()
     dtype = to_torch_dtype(dtype)
-    counts = torch.ones(shape) * convert_to_tensor(counts)
-    probabilities = torch.ones(shape) * convert_to_tensor(probabilities)
+    counts = torch.ones(shape, device=get_device()) * convert_to_tensor(counts)
```
Comment (fchollet) [inline keras/backend/torch/random.py:211]: Most likely we should not use `ones()` and instead use this https://pytorch.org/docs/stable/generated/torch.broadcast_to.html which will be more efficient. `counts = broadcast_to(counts, shape)`, etc.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/backend/torch/random.py (hunk 1 above)
Comment (sampathweb) [reply keras/backend/torch/random.py]: Thanks for the suggestion.  Makes sense. Done

Comment (sampathweb) [reply keras/backend/torch/random.py]: Running GPU Tests to verify they pass.

Comment (sampathweb) [reply keras/backend/torch/random.py]: All tests now pass (CPU and GPU)

**Review by fchollet — APPROVED** [review]
LGTM, thank you

File: keras/backend/torch/random.py (hunk 1 above)
Comment (KhawajaAbaid) [reply keras/backend/torch/random.py]: Yes you're right about that. I used broadcasting in tensorflow but `torch.ones` was being used by existing functions in the `torch/random.py` module so I decided to just go with it.

**Pull request merged**